infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/logreader/LogicalGroupLogEntryReader.java

```diff
@@ -12,10 +34,97 @@
+     * to replicate, the transaction entry will be partially replicated,
+     * avoiding replication of the other streams present in the transaction.
+     *
+     * @param entry transaction stream opaque entry
+     * @return true, if the transaction entry has any valid stream to replicate.
+     * false, otherwise.
+     */
+    @Override
+    boolean isValidTransactionEntry(@NonNull OpaqueEntry entry) {
+        Set<UUID> txEntryStreamIds = new HashSet<>(entry.getEntries().keySet());
+
+        if (txEntryStreamIds.contains(CLIENT_CONFIG_TABLE_ID) &&
+                isCurrentSessionImpacted(entry.getEntries().get(CLIENT_CONFIG_TABLE_ID))) {
+            log.info("Group destination change detected, log entry sync will be stopped and a new snapshot sync " +
+                    "will be triggered！");
+            throw new GroupDestinationChangeException();
+        }
+
+        return super.isValidTransactionEntry(entry);
+    }
+
+    /**
+     * Helper method for checking if the smr entries present in the opaque stream (that current session's log entry
+     * reader is tracking) contain any group destination change with respect to current session. The opaque entries
+     * will be deserialized to verify if the Sink cluster of current session is among the target destinations.
+     *
+     * @param groupConfigTableEntries SMREntries of LogReplicationModelMetadataTable.
+     * @return True if current session is impacted by group destination config change, false otherwise.
+     */
+    private boolean isCurrentSessionImpacted(List<SMREntry> groupConfigTableEntries) {
+        Set<String> groups = ((LogReplicationLogicalGroupConfig) replicationContext.getConfig(session))
+                .getLogicalGroupToStreams().keySet();
+
+        for (SMREntry smrEntry : groupConfigTableEntries) {
+            // Get serialized form of arguments for registry table. They were sent in OpaqueEntry and
+            // need to be deserialized using ProtobufSerializer
+            Object[] objs = smrEntry.getSMRArguments();
+            ByteBuf keyBuf = Unpooled.wrappedBuffer((byte[]) objs[0]);
+            ClientDestinationInfoKey clientInfoKey = (ClientDestinationInfoKey) protobufSerializer
+                    .deserialize(keyBuf, null);
+
+            if (clientInfoKey.getModel() != session.getSubscriber().getModel() ||
+                    !clientInfoKey.getClientName().equals(session.getSubscriber().getClientName())) {
+                // SMREntry for other clients
+                continue;
+            }
+
+            ByteBuf valueBuf = Unpooled.wrappedBuffer((byte[]) objs[1]);
+            CorfuRecord<DestinationInfoVal, Message> destinations =
+                    (CorfuRecord<DestinationInfoVal, Message>) protobufSerializer.deserialize(valueBuf, null);
+
+            // From a session's point of view, there are 2 ways it could be impacted:
+            // (1) a new group is added to have current session's Sink as its destination
```

Comment:
this also includes the case where the Sink corresponding to the current session gets added to an existing group.  Can you add it to the comment as well?

Reply:
actually by "a new group" I was meaning any group that didn't target this Sink before, not only those were not in the client metadata table. Maybe I can just change it to "A group"